src/Castle.Core/DynamicProxy/Generators/MethodFinder.cs

```diff
@@ -78,17 +78,14 @@ private static MethodInfo[] MakeFilteredCopy(MethodInfo[] methodsInCache, Bindin
 
 		private static object RemoveDuplicates(MethodInfo[] infos)
 		{
-			var uniqueInfos = new Dictionary<MethodInfo, object>(MethodSignatureComparer.Instance);
+			var uniqueInfos = new HashSet<MethodInfo>(MethodSignatureComparer.Instance);
 			foreach (var info in infos)
 			{
-				if (!uniqueInfos.ContainsKey(info))
-				{
-					uniqueInfos.Add(info, null);
-				}
+				uniqueInfos.Add(info);
 			}
```

Comment:
This `foreach` loop probably isn't needed at all when using the `HashSet( IEnumerable<T> collection, IEqualityComparer<T> comparer)` ctor instead.

Reply:
Seeing the call site it looks like this method could be removed and the call site just replaced with:

```
type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
    .Distinct(MethodSignatureComparer.Instance)
    .ToArray();
```